File: server/src/main/java/ru/aim/pilot/ui/HomeController.java
package ru.aim.pilot.ui;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import ru.aim.pilot.service.RevisionService;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

@Controller
public class HomeController {

    private final RevisionService revisionService;

    @Autowired
    public HomeController(RevisionService revisionService) {
        this.revisionService = revisionService;
    }

    @RequestMapping("/")
    String index(Model model) {
        model.addAttribute("territories", revisionService.findAllTerritories());
        return "index";
    }

    @RequestMapping(value = "/logout", method = RequestMethod.GET)
    public String logout(HttpServletRequest request, HttpServletResponse response) {
        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
        if (auth != null) {
            new SecurityContextLogoutHandler().logout(request, response, auth);
        }
        return "redirect:/";//You can redirect wherever you want, but generally it's a good practice to show login screen again.
    }
}

File: server/src/main/java/ru/aim/pilot/ui/HomeController.kt
package ru.aim.pilot.ui

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import ru.aim.pilot.service.RevisionService
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse
import javax.servlet.http.HttpSession

@Controller
class HomeController
@Autowired
constructor(private val revisionService: RevisionService) {

    @RequestMapping(value = "/", method = arrayOf(RequestMethod.GET))
    fun index(model: Model, httpSession: HttpSession): String {
        val territoryId = httpSession.getTerritoryId(null)
        /**
         * Если в сессии пользователя не проставлена ссылка на управление
         * значит это админ и отображаем список всех управлений
         * иначе сразу переводим на страницу отделения
         */
        if (territoryId == null) {
            model.addAttribute("territories", revisionService.findAllTerritories())
            return "index"
        } else {
            return "redirect:/revision/list"
        }
    }

    @RequestMapping(value = "/logout", method = arrayOf(RequestMethod.GET))
    fun logout(request: HttpServletRequest, response: HttpServletResponse): String {
        val auth = SecurityContextHolder.getContext().authentication
        if (auth != null) {
            SecurityContextLogoutHandler().logout(request, response, auth)
        }
        return "redirect:/"
    }
}